algolia/region/region.go

```diff
@@ -5,4 +5,5 @@ type Region string
 const (
 	US Region = "us"
 	EU Region = "eu"
```

Comment:
Looks like region `eu` doesn`t exist:

From https://www.algolia.com/doc/rest-api/analytics/
```
The following domains are available:

    United States: https://analytics.us.algolia.com
    Europe (Germany): https://analytics.de.algolia.com
```

Reply:
Yeah, that was the problem. That's why we are introducing this change. I think @aseure is leaving that enumeration to not break peoples code.

Reply:
It's fine to delete it IMO. No one should be using it as it doesn't exist right?

Reply:
@Ant-hem being pragmatic, yes. I agree. Let's delete it. What do you think @aseure ?

Reply:
I'll have to double check that. I think it existed in the past but it was not documented and it triggered the DE cluster. I'll double check with the Analytics team.

Reply:
I just confirmed with @marchelbling and we do have a `recommendation.eu.algolia.com` which prevents us from removing it. And for the record @sfriquet confirmed that the the `eu` region for the analytics endpoint never existed.

@Ant-hem in other words, we'll need to keep it since customers may be using it to instanciate the recommendation client.